crates/noirc_frontend/src/hir/def_collector/dc_mod.rs

```diff
@@ -184,6 +189,43 @@ impl<'a> ModCollector<'a> {
         }
     }
 
+    /// Collect any type aliases definitions declared within the ast.
+    /// Returns a vector of errors if any type aliases were already defined.
+    fn collect_type_aliases(
+        &mut self,
+        type_aliases: Vec<NoirTyAlias>,
+        krate: CrateId,
+        errors: &mut Vec<FileDiagnostic>,
+    ) {
+        for type_alias in type_aliases {
+            let name = type_alias.name.clone();
+
+            // Create the corresponding module for the type aliases namespace
+            let id = match self.push_child_module(&name, self.file_id, false, errors) {
```

Comment:
Why do type aliases need a child module?

I think this code is copied from how structs are handled in which case this was needed as a namespace to store a struct's methods in. Type aliases shouldn't need methods so this can be removed.